File: services/api-AI-support/internal/domain/ekyc.py
from __future__ import annotations

from abc import ABC, abstractmethod
from dataclasses import dataclass
from typing import Sequence

from internal.domain.jobs import AsyncJobHandle


@dataclass(frozen=True)
class BinaryImage:
    content: bytes
    mime_type: str | None = None


@dataclass(frozen=True)
class FaceMatchRequestPayload:
    session_id: str
    ktp_image: BinaryImage
    selfie_image: BinaryImage
    face_match_threshold: float


@dataclass(frozen=True)
class LivenessRequestPayload:
    session_id: str
    liveness_frames: Sequence[BinaryImage]
    gestures: Sequence[str]


class EkycServicePort(ABC):
    @abstractmethod
    def start_face_match(self, payload: FaceMatchRequestPayload) -> AsyncJobHandle:
        raise NotImplementedError

    @abstractmethod
    def start_liveness(self, payload: LivenessRequestPayload) -> AsyncJobHandle:
        raise NotImplementedError

File: services/api-gateway/src/internal/domain/entities/ekyc.rs
#[derive(Clone)]
pub struct BinaryImage {
    pub content: Vec<u8>,
    pub mime_type: Option<String>,
}

#[allow(dead_code)]
pub struct AsyncJobHandle {
    pub job_id: String,
    pub queue: String,
}

pub struct StartFaceMatchJobPayload {
    pub session_id: String,
    pub ktp_image: BinaryImage,
    pub selfie_image: BinaryImage,
    pub face_match_threshold: f64,
}

pub struct StartLivenessJobPayload {
    pub session_id: String,
    pub liveness_frames: Vec<BinaryImage>,
    pub gestures: Vec<String>,
}
